### ISSUE TITLE
Adding custom annotations?

### ISSUE THREAD

**Issue by laukaichung** [reactive-tech/kubegres#14]
I want to use Linkerd and Bank Vault with Kubegres. But `Kubegres` doesn't seem to support custom annotations.  Would you consider this feature?

if adding custom annotations is supported, I'd use Vault to inject secrets into the environment without relying on creating a secret. But right now it looks like replica pods would fail to start without `valueFrom.secretKeyRef`. It keeps asking `Password` in the logs.

````
spec:
  env:
    - name: POSTGRES_PASSWORD
      value: gffdggfdgfdf (Would use #vault:secret/data/path#DB_PASSWORD if custom annotations can be used)

    - name: POSTGRES_REPLICATION_PASSWORD
      value: fgdgfdgdffd
````

**Comment by alex-arica** [comment]
Thank you for your enhancement suggestion. 

When you use:
annotations:
    linkerd.io/inject: enabled

Do you set it from a StatefulSet? Or from another resources so that it is applied to your Pod?

Do you have an example?

**Comment by laukaichung** [comment]
Thank you alex! Looking forward to use this operator. Goodbye Zalando...

**Comment by alex-arica** [comment]
Thanks for your kind feedback about Kubegres.  I agree and we are planning to keep Kubegres simple.

I made a change and manually I tested it locally. I pushed the changes in GIT main so that I can get your feedback.  

Could you please test it for me as well?

Once you confirm that it is working, I am going to update the acceptance tests and create a release.

Please reinstall the Kubegres controller as follows:
```
kubectl apply -f https://raw.githubusercontent.com/reactive-tech/kubegres/main/kubegres.yaml
```

The above installation YAML is directly from GIT main. 

You can add your annotation(s) from the YAML of "kind: Kubegres". 

Please see an example below:

```
kind: Kubegres
metadata:
  name: mypostgres
  namespace: default
  annotations:
    linkerd.io/inject: enabled
```

**Comment by laukaichung** [comment]
I just installed it in a cloud server as well as a local k3d. It works perfectly fine! The master and replicas show the custom annotations. I can even use the env variables directly without creating a secret. 

To upload dumps to S3, I'm planning to use a custom pg images with AWS cli installed and override the backup script. Is that the right approach?

**Comment by alex-arica** [comment]
Thank you for confirming that it is working for you. I am writing a new acceptance test for this change. Once it is done, I am going to create a new release, tag shortly and let you know once it is available. 

In regards to uploads to S3, yes you can use a custom PG image as you suggested. As long as the image is compatible with the official [Postgres docker image](https://hub.docker.com/_/postgres), it should be fine.

You can enable backup in your YAML of "kind: Kubegres", as explained here:
https://www.kubegres.io/doc/enable-backup.html

And before deploying that YAML, you can override the default backup shell script, as per the page here:
https://www.kubegres.io/doc/override-default-configs.html  (please see the example in the section "Override backup_database.sh"). 

I hope that helps.

**Comment by alex-arica** [comment]
I created a new tag 1.4 and release here:
https://github.com/reactive-tech/kubegres/releases

To install Kubegres 1.4 :
```
kubectl apply -f https://raw.githubusercontent.com/reactive-tech/kubegres/v1.4/kubegres.yaml
```

I updated the documentation pages, as follows:
- new version 1.4 added in: [Getting started page](https://www.kubegres.io/doc/getting-started.html)
- Details about "annotations" added in: [All properties explained](https://www.kubegres.io/doc/properties-explained.html)

Please close this ticket if you are happy with the changes.

Thank you for your suggestions!

**Comment by laukaichung** [comment]
Again, thank you very much for this feature! And you are right about installing AWS CLI in the backup script, it should work without the hassle of maintaining a custom image. Will get back to you if it works.

**Comment by laukaichung** [comment]
@alex-arica 
Is it possible to pass those custom annotations to the backup pods as well?
I'm working on a backup script. I just found that the secrets from Vault in the cluster manifest are not able to pass to the backup pods because the backup pods do not share those custom annotations.

**Comment by alex-arica** [comment]
It's a good point!

I have to write it in a new version, because 1.4 was tagged. 

Could you please close this ticket and create a new one?
 
I will make the changes against the new ticket.

**Comment by vincentserpoul** [comment]
@laukaichung having linkerd injected on the backup job makes it hang (known linkerd issue).
Right now, if you add the inject annotation on the kubegress yaml, the backup is broken.

Did you encounter that issue?

There is a new ttlSecondsAfterFinished config for job in k8s 1.21 that seems to be a solution for this. @alex-arica  could we add it to the backup config?
